buildcontext/detectgit.go

```diff
@@ -128,8 +128,8 @@ func detectIsGitDir(ctx context.Context, dir string) error {
 	return nil
 }
 
-// parseGitRemoteURL converts a gitURL like user@host.com:path/to.git or https://host.com/path/to.git to host.com/path/to
-func parseGitRemoteURL(gitURL string) (string, error) {
+// ParseGitRemoteURL converts a gitURL like user@host.com:path/to.git or https://host.com/path/to.git to host.com/path/to
+func ParseGitRemoteURL(gitURL string) (string, error) {
```

Comment:
Seems like this no longer belongs to this package since it's being used in other contexts too. Perhaps the entire file `detectgit.go` can be moved to something like `gitutil` package?

Reply:
good point, want to take another look after the move?